crates/apub/src/activities/deletion/mod.rs

```diff
@@ -150,20 +144,12 @@ async fn verify_delete_activity_post_or_comment(
   Ok(())
 }
 
-struct WebsocketMessages {
-  community: UserOperationCrud,
-  post: UserOperationCrud,
-  comment: UserOperationCrud,
-}
-
 /// Write deletion or restoring of an object to the database, and send websocket message.
 /// TODO: we should do something similar for receive_remove_action(), but its much more complicated
 ///       because of the mod log
 async fn receive_delete_action(
   object: &Url,
   actor: &ObjectId<ApubPerson>,
-  ws_messages: WebsocketMessages,
-  deleted: bool,
```

Comment:
I'm 99% sure I broke undeletes here, but I'm not sure the best way to refactor it. Your `delete`  trait function can only delete, not undelete. It'd have to take in a bool to do that.

Reply:
Damn, thats true. Then we cant do it this way, and you need to revert the commit :(

Reply:
Would it be better to add a `deleted: bool` to the delete trait function, or maybe add an undelete function? I'm confused as to why delete exists but undelete doesn't.

Reply:
The delete function is necessary to handle Tombstones when fetching. And the library is supposed to work for other projects as well, so i dont think it makes sense to add support for something Lemmy specific like Undo/Delete. Lemmy specific in the sense that its perfectly possible to have an apub project without that functionality, so such a parameter would just be useless.